src/test/java/org/javarosa/core/model/RepeatTest.java

```diff
@@ -73,4 +73,35 @@ public void whenRepeatRelevanceIsDynamic_andNotRelevant_repeatPromptIsSkipped()
         int event = scenario.next();
         assertThat(event, is(FormEntryController.EVENT_END_OF_FORM));
     }
+
+    @Test
+    public void whenRepeatAndTopLevelNodeHaveSameRelevanceExpression_andExpressionEvaluatesToFalse_repeatPromptIsSkipped() throws Exception {
```

Comment:
This problem **is** unique to repeats right? I guess with normal nodes the problem just wouldn't matter?